core/aws-lsp-fqn/src/common/CommonFqnWorkerPool.ts

```diff
@@ -0,0 +1,44 @@
+import { pool, Pool } from 'workerpool'
+import { DEFAULT_MAX_WORKERS, DEFAULT_TIMEOUT, FQN_WORKER_ID } from './defaults'
+import { IFqnWorkerPool } from './IFqnWorkerPool'
+import { ExtractorResult, FqnExtractorInput, Logger, WorkerPoolConfig } from './types'
+
+export class CommonFqnWorkerPool implements IFqnWorkerPool {
+    #workerPool: Pool
+    #logger?: Logger
+    #timeout: number
+
+    constructor(filePath: string, { timeout = DEFAULT_TIMEOUT, logger, workerPoolOptions }: WorkerPoolConfig = {}) {
+        this.#timeout = timeout
+        this.#logger = logger
+        this.#workerPool = pool(filePath, {
+            maxWorkers: DEFAULT_MAX_WORKERS,
```

Comment:
Should we also limit `maxQueueSize`?

Reply:
Yes we should, thanks! I set it to 10 for now